.github/actions/check-pr-semver-labels/action.yml

```
@@ -0,0 +1,136 @@
+name: Check PR semantic versioning labels
+description: "Check semantic versioning labels"
+
+inputs:
+  github_token:
+    required: true
+    description: The github token used to read PR metadata
+
+outputs:
+  semver_app:
+    description: "Application semantic version label"
+    value: ${{ steps.check_labels.outputs.semver_app }}
+  semver_chart:
+    description: "Chart semantic version label"
+    value: ${{ steps.check_labels.outputs.semver_chart }}
+  skip_release:
+    description: "Skip release flag"
+    value: ${{ steps.check_labels.outputs.skip_release }}
+runs:
+  using: "composite"
+  steps:
+    - name: Retrieve semantic versioning labels on PR
+      id: check_labels
+      uses: actions/github-script@v6.3.3
+      env:
+        VALID_APP_VERSION_LABELS: patch, minor, major, ignore-for-release
+        VALID_CHART_VERSION_LABELS: chart-patch, chart-minor, chart-major
+        SKIP_RELEASE_LABEL: skip-release
+      with:
+        github-token: ${{ inputs.github_token }}
+        script: |
+          const { VALID_APP_VERSION_LABELS, VALID_CHART_VERSION_LABELS, SKIP_RELEASE_LABEL } = process.env
+          const validLabelsSemver = VALID_APP_VERSION_LABELS.trim().split(',').map(word => word.trim()).filter(word => word.length > 0);
+          const validLabelsChartSemver = VALID_CHART_VERSION_LABELS.trim().split(',').map(word => word.trim()).filter(word => word.length > 0);
+          const skipReleaseLabel = SKIP_RELEASE_LABEL.trim();
+          var issueNumber = context.issue.number;
+          if (issueNumber === undefined) {
+            core.info('Retrieving issue number from merge commit message')
+            var mergeCommit = await github.rest.git.getCommit({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              commit_sha: context.sha,
+            });
+            if (!mergeCommit) {
+              throw `Could not find commit for sha ${context.sha}`;
+            }
+            const commitMessage = mergeCommit.data.message;
+            core.info(`Commit message: ${commitMessage}`);
+            const extractedPrNumber = commitMessage.match('\\(#(\\d{1,9})\\)'); 
```

Doesn't this match only PRs with number < 10?

---

No, this is a greedy quantifier, so it match all pr from #1 to #999999999

---

Oh right, I didn't see `{1,9}` correctly

---

Btw, you can use regex literals to improve readability a bit:
```suggestion
            const extractedPrNumber = commitMessage.match(/\(#(\d{1,9})\)/); 
```